lib/cpp/src/thrift/transport/TSSLSocket.cpp

```diff
@@ -140,13 +145,11 @@ SSLContext::SSLContext(const SSLProtocol& protocol) {
   if(protocol == SSLTLS)
   {
     ctx_ = SSL_CTX_new(SSLv23_method());
-  }
-  else if(protocol == SSLv3)
-  {
+#ifndef OPENSSL_NO_SSL3
```

Comment:
Please use separate commit for this change. I guess it's not related to CMake.

Reply:
I guess not but it does relate to default version of OpenSSL in Hunter. Seems to be older versions don't need this fix according to this pull request: https://github.com/apache/thrift/pull/1102.

Reply:
> I guess not but it does relate to default version of OpenSSL in Hunter

Not sure I follow. It's not about CMake or Hunter. This change is about C++ code. Make a separate commit with this change.

> according to this pull request: apache#1102

Please add this link to commit too. Effectively it means we have two commits here.